plugins/woocommerce/src/Blocks/Domain/Services/CheckoutLink.php

```diff
@@ -94,21 +94,71 @@ protected function get_products_from_checkout_link() {
 		$raw_products = array_filter( explode( ',', wc_clean( wp_unslash( $_GET['products'] ?? '' ) ) ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 		$products     = [];
 
-		foreach ( $raw_products as $product_id_qty ) {
-			if ( strpos( $product_id_qty, ':' ) !== false ) {
-				list( $product_id, $qty ) = explode( ':', $product_id_qty );
+		foreach ( $raw_products as $raw_product ) {
+			$cart_item_data = [];
+
+			if ( strpos( $raw_product, ':' ) !== false ) {
+
+				list( $product_id, $product_data ) = explode( ':', $raw_product );
```

Comment:
We should keep the original schema intact, so if you need to add `id:qty:data` (data being optional) in that way to keep the basic links working.